### PR TITLE
[Dataset] Remove support for `DatasetDict` as input into `from_huggingface()`

### LINKED ISSUE #37523: [Data] Only support Hugging Face dataset in `from_huggingface` API
### Description

Currently `form_huggingface` accepts both Hugging Face `datasets.Dataset` and `datasets.DatasetDict`. When user is passing a `datasets.DatasetDict`, the below warning would throw:

```
>>> ray_datasets = ray.data.from_huggingface(current_dataset)
2023-07-17 12:50:58,330 WARNING read_api.py:1856 -- You provided a Huggingface DatasetDict which contains multiple datasets. The output of `from_huggingface` is a dictionary of Ray Datasets. To convert just a single Huggingface Dataset to a Ray Dataset, specify a split. For example, `ray.data.from_huggingface(my_dataset_dictionary['train'])`. Available splits are ['train', 'validation', 'test'].
```

This would be confusing to users, and the current `from_huggingface` API is not consisent with other `from_xxx` APIs, which do not allow overloading with dict.

In this issue, we can change `from_huggingface` API to just accept `datasets.Dataset`

```
def from_huggingface(
    dataset: "datasets.Dataset",
)
```

In addition, raise an exception with instruction for users if they pass a dictionary.

### Use case

As above.

### PR DESCRIPTION
**Pull request by scottjlee**
## Why are these changes needed?

To keep the `from_huggingface()` API simple and consistent with other `from_xxx` read APIs, we remove support for Hugging Face `datasets.DatasetDict`, so that the method only accepts a Hugging Face `datasets.Dataset`.

## Related issue number

Closes #37523

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/data/read_api.py [hunk 1]
```diff
@@ -2122,42 +2094,27 @@ def from_huggingface(
             )
 
     Args:
-        dataset: A `Hugging Face Datasets Dataset`_ or `DatasetDict`_.
-            :class:`~ray.data.IterableDataset` isn't supported.
+        dataset: A `Hugging Face Datasets Dataset`_.
+            :class:`~ray.data.IterableDataset` and
```
Comment (amogkam) [inline python/ray/data/read_api.py:2098]: Man this IterableDataset link is just straight up wrong 😅. Not sure how this had slipped in before.

File: python/ray/data/read_api.py [hunk 2]
```diff
@@ -2122,42 +2094,27 @@ def from_huggingface(
...
         # mapping from the transformations.
-        hf_ds_arrow = ds.with_format("arrow")
+        hf_ds_arrow = dataset.with_format("arrow")
         ray_ds = from_arrow(hf_ds_arrow[:])
         return ray_ds
-
-    if isinstance(dataset, datasets.DatasetDict):
-        available_keys = list(dataset.keys())
-        logger.warning(
-            "You provided a Huggingface DatasetDict which contains multiple "
```
Comment (amogkam) [inline python/ray/data/read_api.py:2146]: Let's keep this, but change to an error message

Comment (scottjlee) [reply python/ray/data/read_api.py:2146]: would the last case raising `TypeError` cover this case as well? or do we want to explicitly raise a separate error for this case, with a message like "DatasetDict is not supported"?

File: python/ray/data/read_api.py [hunk 3]
```diff
@@ -2122,42 +2094,38 @@ def from_huggingface(
...
-    if isinstance(dataset, datasets.DatasetDict):
+    elif isinstance(dataset, datasets.DatasetDict):
         available_keys = list(dataset.keys())
-        logger.warning(
-            "You provided a Huggingface DatasetDict which contains multiple "
-            "datasets. The output of `from_huggingface` is a dictionary of Ray "
-            "Datasets. To convert just a single Huggingface Dataset to a "
+        logger.error(
+            "You provided a Hugging Face DatasetDict which contains multiple "
+            "datasets, but `from_huggingface` now only accepts a siingle Hugging Face "
```
Comment (c21) [inline python/ray/data/read_api.py:2119]: `siingle` -> `single`

**Review by c21 — APPROVED** [review]

File: python/ray/data/read_api.py [hunk 4]
```diff
@@ -2122,42 +2094,38 @@ def from_huggingface(
...
         return ray_ds
-
-    if isinstance(dataset, datasets.DatasetDict):
+    elif isinstance(dataset, datasets.DatasetDict):
         available_keys = list(dataset.keys())
-        logger.warning(
-            "You provided a Huggingface DatasetDict which contains multiple "
-            "datasets. The output of `from_huggingface` is a dictionary of Ray "
-            "Datasets. To convert just a single Huggingface Dataset to a "
+        logger.error(
```
Comment (amogkam) [inline python/ray/data/read_api.py:2117]: don't we want to raise an exception here?

**Review by amogkam — CHANGES_REQUESTED** [review]

File: python/ray/data/read_api.py (hunk 4 above)
Comment (scottjlee) [reply python/ray/data/read_api.py:2117]: ah okay, got it. let me raise a ValueError here

File: python/ray/data/read_api.py (hunk 2 above)
Comment (ericl) [reply python/ray/data/read_api.py:2146]: Btw, it should be a DeprecationWarning raised since technically this is a breaking change right.

**Review by amogkam — APPROVED** [review]

**Pull request merged**